Partial Word Count (8% accuracy, 100 samples) - accurately counting words in a specified section of the user's message

michaeljelly opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. We encourage partial PR's with ~5-10 example that we can then run the evals on and share the results with you so you know how your eval does with GPT-4 before writing all 100 examples.

## Eval details 📑
### Eval name
word_count

### Eval results
[2023-03-17 18:05:41,924] [oaieval.py:209] Final report:
[2023-03-17 18:05:41,925] [oaieval.py:211] accuracy: 0.08
[2023-03-17 18:05:41,925] [oaieval.py:211] f1_score: 0.05333333333333333

### Eval description

This eval tests whether the model is able to accurately count the number of words in an indicated part of the user's message.

### What makes this a useful eval?

When instructing a model to do certain tasks, it is useful to be able to have fine grained control over the length of the output.  Many places humans write text, such as government forms, password fields require specific lengths of text to be written. 

The LLM struggles to follow instructions to write a piece of text X words long. With no sense of how long a piece of text is even at very small numbers of words, this is to be expected! This eval tries to test the model's understanding of how long a piece of text is, specifically a subset of the user's message that is the answer to the question.

It also seems like a very basic piece of functionality but one that LLMs are particularly bad at.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] Include at least 100 high quality examples (it is okay to only contribute 5-10 meaningful examples and have us test them with GPT-4 before adding all 100)

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.jsonl`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields in the evals PR form
- [x] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"ideal":["5"],"input":[{"role":"system","content":"Count the number of words in the answer to the question."},{"role":"user","content":"What are renewable energy sources? Clean, sustainable power generation options"}]}
{"ideal":["9"],"input":[{"role":"system","content":"Count the number of words in the answer to the question."},{"role":"user","content":"What is the theory of relativity? Einstein's physics concept, relates gravity, space, time, and motion"}]}
{"ideal":["15"],"input":[{"role":"system","content":"Count the number of words in the answer to the question."},{"role":"user","content":"What is quantum mechanics? Branch of physics studying microscopic particles, their behavior and interactions at atomic and subatomic levels"}]}
{"ideal":["14"],"input":[{"role":"system","content":"Count the number of words in the answer to the question."},{"role":"user","content":"What is the purpose of the United Nations? International organization promoting peace, security, cooperation, and development; addresses global issues and facilitates diplomacy"}]}
{"ideal":["19"],"input":[{"role":"system","content":"Count the number of words in the answer to the question."},{"role":"user","content":"What is climate change? Long-term alteration of Earth's climate patterns, attributed to increased greenhouse gas emissions; results in extreme weather, rising sea levels"}]}
  ```
</details>

MrDevel0per commented:
Is this a duplicate of #96 or #275?

michaeljelly commented:
No it's not, the success rate is much lower. Two differences are: I'm asking it to count just a specific part of the user's message, and I'm using the system message to instruct it rather than the user message.

MrDevel0per commented:
Got it - this seems like a good eval then! I see it ends up testing not only the model's ability to detect word count, but also segment the user input.

michaeljelly commented:
Thanks! Just updated the title to be more specific and added the accuracy details to the pull request's title. You can see it only gets 8% accuracy vs the 40% that the other evaluation got!

luqman-openai reviewed (changes requested):
Thanks for submitting this eval. I would like to recommend some changes.

- The output format is not specified in the prompt. There is no guarantee that model will always return an integer, it can response with correct answer by printing `9` as `Nine` which will be considered correct if the response format is not specified. Add proper instructions in the prompt to output just an integer without any additional text.

- The `Match` evalutor is recommened in this case instead of `FuzzyMatch`

- Push the sample input data to the Git LFS, all the sample files are hosted there.

We would love to review the PR again after the above mentioned points is addressed.

luqman-openai commented:
Closing for now, please feel free to reopen if you get a chance to address the comments.
